RecoLocalCalo/HGCalRecProducers/src/ComputeClusterTime.cc

```diff
@@ -0,0 +1,126 @@
+#include "RecoLocalCalo/HGCalRecProducers/interface/ComputeClusterTime.h"
+
+// functions to select the hits to compute the time of a given cluster
+// start with the only hits with timing information
+// average among the hits contained in the chosen time interval
+
+// N.B. time is corrected wrt vtx-calorimeter distance
+// with straight line and light speed hypothesis
+// for charged tracks or heavy particles (longer track length or beta < 1)
+// need to correct the offset at analysis level
+
+using namespace hgcalsimclustertime;
+
+std::vector<size_t> decrease_sorted_indices(const std::vector<float>& v) {
+  // initialize original index locations
+  std::vector<size_t> idx(v.size());
+  std::iota(idx.begin(), idx.end(), 0);
+
+  // sort indices based on comparing values in v (decreasing order)
+  std::sort(idx.begin(), idx.end(), [&v](size_t i1, size_t i2) { return v[i1] < v[i2]; });
+  return idx;
+};
+
+ComputeClusterTime::ComputeClusterTime(float Xmin, float Xmax, float Cterm, float Aterm)
+    : _Xmin(Xmin), _Xmax(Xmax), _Cterm(Cterm), _Aterm(Aterm) {
+  if (_Xmin < 0)
```

Comment:
```_Xmin = 0``` is still allowed here in principle (at the contrary to what coded in the ```setParameters``` method below). This would generate a division by 0 when calling ```timeResolution```